src/toil/job.py

```diff
@@ -157,11 +159,65 @@ def _parseResource(name, value):
                             % (name, type(value)))
 
 
-class Job(ResourceRequirementMixin):
+class JobLikeObject(ResourceRequirementMixin):
+    def __init__(self, name, job=None, **requirements):
+        super(JobLikeObject, self).__init__(**requirements)
+        self.name = name
+        self.job = job or self.__class__.__name__
+
+    def __str__(self):
+        printedName = '\''+self.job + '\''
+        if self.name:
```

Comment:
With the same reasoning for disliking how the constructor treats `None` and `""` the same, I think if someone passes and empty string for `name`, that should be reflected in the output.